src/testing/mod.rs

```diff
@@ -33,22 +32,23 @@ use tempdir::TempDir;
 // Loader for using in tests.  Just creates unit metadata and defualt Keytree.
 pub struct TrivialKeystoreLoader {
     pub dir: PathBuf,
+    rng: ChaChaRng,
 }
 
 impl<L: Ledger> KeystoreLoader<L> for TrivialKeystoreLoader {
-    type Meta = LoaderMetadata;
+    type Meta = ();
 
     fn location(&self) -> PathBuf {
         self.dir.clone()
     }
 
-    fn create(&mut self) -> Result<(LoaderMetadata, KeyTree), KeystoreError<L>> {
-        let key = KeyTree::from_password_and_salt(&[], &[0; 32]).context(KeySnafu)?;
+    fn create(&mut self) -> Result<((), KeyTree), KeystoreError<L>> {
+        let key = KeyTree::random(&mut self.rng).0;
         Ok((Default::default(), key))
     }
 
-    fn load(&mut self, _meta: &mut LoaderMetadata) -> Result<KeyTree, KeystoreError<L>> {
-        KeyTree::from_password_and_salt(&[], &[0; 32]).context(KeySnafu)
+    fn load(&mut self, _meta: &mut ()) -> Result<KeyTree, KeystoreError<L>> {
+        Ok(KeyTree::random(&mut self.rng).0)
```

Comment:
This will create a different `KeyTree` on load than was used on `create`. Probably better to do `type Meta = KeyTree`, and store the generated `KeyTree` in `create`. Sorry I missed this last time.

Reply:
I decided to store the `ChaChaRng` that we use to create the tree as `Meta` and use it to recreate the tree in `load()`.  I think is should be equivalent and prevents us from having to implement `Serialize`, `Deserialize`, and `PartialEq` for `KeyTree`.